planning/MartenV4.md

```diff
@@ -0,0 +1,114 @@
+# Marten v4 Ideas
+
+## Pulling it Off
+
+We've got the typical problems of needing to address incoming pull requests and bug issues in master while probably needing to have a long lived branch for *v4*. 
+
+Here are some thoughts:
+
+* First do a bug sweep 3.12 release to address as many of the tactical problems as we can before branching to v4?
+* Start with the unit testing improvements as a way to speed up the build before we dive into much more of this? 
+* Possibly do a series of v4, then v5 releases to do this in smaller chunks? We've mostly said do the event store as v4, then Linq improvements as v5
+* Do the Event Store v4 work in a separate project built as an add on from the very beginning, but leave the existing event store in place. That would enable us to do a lot of work *and* mostly be able to keep that work in master so we don't have long-lived branch problems
+* Damn the torpedos and full speed ahead!
+
+## Miscellaneous Ideas
+
+* Look at some kind of object pooling for the `DocumentSession` / `QuerySession` objects?
+* Ditch the document by document type schema configuration. Do that, and I think we open the door for multi-tenancy by schema
```

Comment:
That also needs more clarifications for me.

Reply:
Today you can have Marten put document "A" in schema "A", then document "B" in schema "B" and so on. I want to get rid of that if possible as a way to do "multi-tenancy via schemas"

Reply:
Understood. Would be nice to include that (and other clarifications from the comments into document).